server/mdm/apple/apple_mdm.go

```diff
@@ -212,23 +214,34 @@ func (d *DEPService) RegisterProfileWithAppleDEPServer(ctx context.Context, team
 	jsonProf.AwaitDeviceConfigured = true
 
 	depClient := NewDEPClient(d.depStorage, d.ds, d.logger)
-	res, err := depClient.DefineProfile(ctx, DEPName, &jsonProf)
+	// Get all relevant org names
+	var tmID *uint
+	if team != nil {
+		tmID = &team.ID
+	}
+	orgNames, err := d.ds.GetABMTokenOrgNamesForTeam(ctx, tmID)
 	if err != nil {
-		return ctxerr.Wrap(ctx, err, "apple POST /profile request failed")
+		return ctxerr.Wrap(ctx, err, "getting org names for team to register profile")
 	}
 
-	if setupAsst != nil {
-		setupAsst.ProfileUUID = res.ProfileUUID
-		if err := d.ds.SetMDMAppleSetupAssistantProfileUUID(ctx, setupAsst.TeamID, res.ProfileUUID); err != nil {
-			return ctxerr.Wrap(ctx, err, "save setup assistant profile UUID")
-		}
-	} else {
-		var tmID *uint
-		if team != nil {
-			tmID = &team.ID
+	if len(orgNames) == 0 {
+		orgNames = append(orgNames, DEPName)
+	}
```

Comment:
in which scenarios do you think this could happen? ideally we want to completely remove `DEPName`

Reply:
This situation comes up in the test when assigning to a team that has no hosts. Previously, that assignment could happen even if there were no hosts, because we just used the DEPName. But now, we get the org names via the hosts. 

This is basically a hack to allow a team with no hosts to get the profile.

Reply:
my hunch is that we shouldn't do this in that case. Instead, use the token assigned to the team (related to the other discussion we're having) or not assign anything at all.

Reply:
I'm thinking, to preserve existing behavior, but also make it correct: can we `JOIN` with `abm_tokens` in `GetABMTokenOrgNamesForTeam`?

Reply:
@roperzh 
> can we JOIN with abm_tokens in GetABMTokenOrgNamesForTeam?

Not sure what you mean; do you mean use the default macos team column as a way to see if the token is related to this team if no hosts exist?

Reply:
@jahzielv what I'm thinking is, the full set of tokens for a team is the union of the sets:

1. Any row in `abm_tokens` that references the team
2. Any token referenced by a host that belongs to the team

Seems like we're currently covering `2`, but we ignore all tokens if they're not referenced by a host? it might be fine either way, `1` is only if you want to preserve existing behavior (ie: we define profiles even if you don't have hosts in that team)

Reply:
@roperzh  Yeah, that makes sense.

> 1 is only if you want to preserve existing behavior (ie: we define profiles even if you don't have hosts in that team)

To me it seems OK to just skip defining the profile for a team with no hosts, but I'm definitely not the expert on this system! If that seems OK to you, then I will just make `RegisterProfileWithAppleDEPServer` do nothing if there are no hosts in the team.